packages/contracts/contracts/MarketLiquidityRewards.sol

```diff
@@ -0,0 +1,462 @@
+        );
+
+        _verifyExpectedTokenAddress(
+            collateralTokenAddress,
+            allocatedReward.requiredCollateralTokenAddress
+        );
+
+        require(
+            marketId == allocatedRewards[_allocationId].marketId,
+            "MarketId mismatch for allocation"
+        );
+
+        uint256 principalTokenDecimals = IERC20MetadataUpgradeable(
+            principalTokenAddress
+        ).decimals();
+
+        address rewardRecipient = _verifyAndReturnRewardRecipient(
+            allocatedReward.allocationStrategy,
+            bidState,
+            borrower,
+            lender
+        );
+
+        uint256 amountToReward = _calculateRewardAmount(
+            principalAmount,
+            principalTokenDecimals,
+            allocatedReward.rewardPerLoanPrincipalAmount
+        );
+
+        //transfer tokens reward to the msgsender
+        IERC20Upgradeable(allocatedRewards[_allocationId].rewardTokenAddress)
+            .transfer(rewardRecipient, amountToReward);
+
+        _decrementAllocatedAmount(_allocationId, amountToReward);
+
+        emit ClaimedRewards(
+            _allocationId,
+            _bidId,
+            rewardRecipient,
+            amountToReward
+        );
+    }
+
+    /**
+     * @notice Verifies that the bid state is appropriate for claiming rewards based on the allocation strategy and then returns the address of the reward recipient(borrower or lender)
+     * @param _strategy - The strategy for the reward allocation.
+     * @param _bidState - The bid state of the loan.
+     * @param _borrower - The borrower of the loan.
+     * @param _lender - The lender of the loan.
+     * @return rewardRecipient_ The address that will receive the rewards. Either the borrower or lender.
+     */
+    function _verifyAndReturnRewardRecipient(
+        AllocationStrategy _strategy,
+        BidState _bidState,
+        address _borrower,
+        address _lender
+    ) internal virtual returns (address rewardRecipient_) {
+        if (_strategy == AllocationStrategy.BORROWER) {
+            require(_bidState == BidState.PAID, "Invalid bid state for loan.");
+
+            rewardRecipient_ = _borrower;
+        } else if (_strategy == AllocationStrategy.LENDER) {
+            //Loan must have been accepted in the past
+            require(
+                _bidState >= BidState.ACCEPTED,
+                "Invalid bid state for loan."
+            );
+
+            rewardRecipient_ = _lender;
+        } else {
+            revert("Unknown allocation strategy");
+        }
+    }
+
+    /**
+     * @notice Decrements the amount allocated to keep track of tokens in escrow
+     * @param _allocationId - The id for the allocation to decrement
+     * @param _amount - The amount of ERC20 to decrement
+     */
+    function _decrementAllocatedAmount(uint256 _allocationId, uint256 _amount)
+        internal
+    {
+        allocatedRewards[_allocationId].rewardTokenAmount -= _amount;
+    }
```

Comment:
seems silly to have an internal function with 1 line only be used once in the codebase. 

either use this in the `deallocateRewards` function as well or inline the code in the `claimRewards`

Reply:
okay now this is also being used in deallocate